app/Service/Student/ModalityService.php

```
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Student;;
+
+use App\Models\Resume;
+
+class ModalityService
+{
+    public function execute($studentId)
+    {
+
+        $modality = $this->getModalityByStudentId($studentId);
+
+        if (!$modality) {
+            throw new \Exception('No se encontró el currículum del usuario', 404);
```

You should throw a controlled exception and catch it in the controller